File: Backend/config/deployment.py
"""Mode de déploiement : `local` (serveur on-premise) ou `aws` (EC2 + S3).

DEPLOYMENT_MODE=local  → stockage fichiers sous LOCAL_DATA_ROOT, aucun appel AWS.
DEPLOYMENT_MODE=aws    → comportement historique (S3, EC2, IAM).

Lu à chaque appel (pas de cache) pour que les tests puissent basculer de mode.
"""

import os
from pathlib import Path

from dotenv import load_dotenv

load_dotenv()

LOCAL = "local"
AWS = "aws"


def deployment_mode() -> str:
    mode = os.getenv("DEPLOYMENT_MODE", LOCAL).strip().lower()
    return AWS if mode == AWS else LOCAL


def is_local() -> bool:
    return deployment_mode() == LOCAL


def local_data_root() -> Path:
    """Racine des données locales (référence, patients, modèles, tmp)."""
    return Path(os.getenv("LOCAL_DATA_ROOT", "/data/zaynb")).resolve()

File: Frontend/lib/deployment.ts
/**
 * Mode de déploiement (build-time) : `local` (serveur on-premise, défaut) ou `aws`.
 * Ne change que les libellés et la validation des chemins ; l'API reste la même.
 */
export const DEPLOYMENT_MODE: 'local' | 'aws' =
  process.env.NEXT_PUBLIC_DEPLOYMENT_MODE === 'aws' ? 'aws' : 'local';

export const IS_LOCAL = DEPLOYMENT_MODE === 'local';

/** Racine des données sur le serveur (affichage des exemples uniquement). */
export const LOCAL_DATA_ROOT =
  process.env.NEXT_PUBLIC_LOCAL_DATA_ROOT || '/data/zaynb';

export const INPUT_LOCATION_LABEL = IS_LOCAL ? 'chemin serveur' : 'S3';
